client/state/shortcodes/actions.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import { omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	SHORTCODE_RECEIVE,
+	SHORTCODE_REQUEST,
+	SHORTCODE_REQUEST_FAILURE,
+	SHORTCODE_REQUEST_SUCCESS
+} from 'state/action-types';
+
+export function fetchShortcode( siteId, shortcode ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SHORTCODE_REQUEST,
+			siteId,
+			shortcode
+		} );
+
+		return wpcom.undocumented().site( siteId ).shortcodes( { shortcode } ).then( ( data ) => {
+			dispatch( {
+				type: SHORTCODE_REQUEST_SUCCESS,
+				siteId,
+				shortcode
+			} );
+
+			dispatch( {
+				type: SHORTCODE_RECEIVE,
+				siteId,
+				shortcode,
+				data: {
```

Comment:
Couple thoughts:

- I know it's how it was previously, but is there value in changing the structure to reassign `result` as `body` ? Would simplify here to simply dispatch `data`.
- If we were to continue transforming the structure, do you think this should be done in the reducer instead?